specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json

```diff
@@ -0,0 +1,1712 @@
+          },
+          {
+            "$ref": "#/parameters/BatchJobIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/EndpointNameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BatchJobResource"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "BatchJobDeployment"
+        ],
+        "summary": "Delete Batch Inference job deployment.",
+        "operationId": "BatchJobsDeployments_Delete",
+        "x-ms-examples": {
+          "Delete BatchJobDeployment": {
+            "$ref": "./examples/deleteBatchJobDeployment.json"
+          }
+        },
+        "produces": [
+          "text/plain",
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/EndpointDeploymentIdParameter"
+          },
+          {
+            "$ref": "#/parameters/BatchJobIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/EndpointNameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          },
+          "200": {
+            "description": "Success"
+          },
+          "204": {
+            "description": "Success"
+          }
+        }
+      }
+    },
+    "/mferp/managementfrontend/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/batchEndpoints/{endpointName}/deployments/{deploymentId}/jobs": {
+      "post": {
```

Comment:
jobs should be created via a PUT on /jobs/{jobname}, not a POST on /jobs

Reply:
The identifiers here are service generated, hence I have a 'location:' in the output added as per the guidelines [here](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#741-post).

Reply:
That is not supported in ARM.  The docs you linked to are REST API guidelines, ARM has its own restrictions.  Resources should be created with PUT calls (and thus creatable via template deployments).  You can have a read-only service generated "uniqueId" property within the resource's properties if necessary.

Reply:
Switching to use a PUT